hoot-core/src/main/cpp/hoot/core/algorithms/splitter/CornerSplitter.cpp

```diff
@@ -100,40 +116,159 @@ void CornerSplitter::splitCorners()
         double h2 = atan2(next.getCoordinate().y - current.getCoordinate().y,
                           next.getCoordinate().x - current.getCoordinate().x);
 
-        double threshold = toRadians(55.0);
         double delta = fabs(h2-h1);
 
         if (delta > M_PI)
           delta = twopi - delta;
 
-
         // If we make enough of a turn, split the way
         if (delta > threshold)
         {
           LOG_TRACE("splitting way with delta: " << delta);
-          _splitWay(pWay->getId(), nodeIdx, pWay->getNodeId(nodeIdx));
-          split = true;
+          split = _splitWay(pWay->getId(), nodeIdx, pWay->getNodeId(nodeIdx));
+        }
+      }
+    }
+  }
+  //  Try out the rounded corner splitting algorithm if desired
+  if (_splitRounded)
+    _splitRoundedCorners();
+}
+
+void CornerSplitter::_splitRoundedCorners()
+{
```

Comment:
Of all the changes that you've already reviewed, this is the function that contains the splitting algorithm for rounded corners.  Essentially what it does is takes a total change in heading between way segments of the sliding window of size `corner.splitter.rounded.max.node.count` (defaults to 6).  If it finds a window that deviates more than `corner.splitter.rounded.threshold` (defaults to 55 degrees) then a point is selected within that window where the change in heading exceeds half of the total change to split the way.  In the unit test, the dataset contains a "U" shaped way that is split twice, once at each "corner" making three ways that are more conflate-able with ways that meet at right angles in this case.